src/common/time/src/duration.rs

```diff
@@ -0,0 +1,292 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::cmp::Ordering;
+use std::fmt::{Display, Formatter};
+use std::hash::{Hash, Hasher};
+
+use serde::{Deserialize, Serialize};
+
+use crate::timestamp::TimeUnit;
+
+/// [Duration] represents the elapsed time in either seconds, milliseconds, microseconds or nanoseconds.
+#[derive(Debug, Clone, Default, Copy, Serialize, Deserialize)]
+pub struct Duration {
+    value: i64,
+    unit: TimeUnit,
+}
+
+impl Duration {
+    /// Create a new Duration with value and TimeUnit.
+    pub fn new(value: i64, unit: TimeUnit) -> Self {
+        Self { value, unit }
+    }
+
+    /// Create a new Duration in second.
+    pub fn new_second(value: i64) -> Self {
+        Self {
+            value,
+            unit: TimeUnit::Second,
+        }
+    }
+
+    /// Create a new Duration in millisecond.
+    pub fn new_millisecond(value: i64) -> Self {
+        Self {
+            value,
+            unit: TimeUnit::Millisecond,
+        }
+    }
+
+    /// Create a new Duration in microsecond.
+    pub fn new_microsecond(value: i64) -> Self {
+        Self {
+            value,
+            unit: TimeUnit::Microsecond,
+        }
+    }
+
+    /// Create a new Duration in nanosecond.
+    pub fn new_nanosecond(value: i64) -> Self {
+        Self {
+            value,
+            unit: TimeUnit::Nanosecond,
+        }
+    }
+
+    /// Return the TimeUnit of current Duration.
+    pub fn unit(&self) -> TimeUnit {
+        self.unit
+    }
+
+    /// Return the value of current Duration.
+    pub fn value(&self) -> i64 {
+        self.value
+    }
+
+    /// Split a [Duration] into seconds part and nanoseconds part.
+    /// Notice the seconds part of split result is always rounded down to floor.
+    fn split(&self) -> (i64, u32) {
+        let sec_mul = (TimeUnit::Second.factor() / self.unit.factor()) as i64;
+        let nsec_mul = (self.unit.factor() / TimeUnit::Nanosecond.factor()) as i64;
+
+        let sec_div = self.value.div_euclid(sec_mul);
+        let sec_mod = self.value.rem_euclid(sec_mul);
+        // safety:  the max possible value of `sec_mod` is 999,999,999
+        let nsec = u32::try_from(sec_mod * nsec_mul).unwrap();
+        (sec_div, nsec)
+    }
+}
+
+// Convert i64 to Duration Type.
+// Default TimeUnit is Millisecond.
+impl From<i64> for Duration {
+    fn from(v: i64) -> Self {
+        Self {
+            value: v,
+            unit: TimeUnit::Millisecond,
+        }
+    }
+}
+
+// return i64 value of Duration.
+impl From<Duration> for i64 {
```

Comment:
Maybe you also need to implement the conversion between `common_time::Duration` and `std::time::Duration`

Reply:
thx, I will implement it(•̀ᴗ• )